spring-integration-core/src/main/java/org/springframework/integration/splitter/AbstractMessageSplitter.java

```diff
@@ -116,6 +123,26 @@ else if (result instanceof Iterator<?>) {
 		return builder;
 	}
 
+	/**
+	 * Return true if the subclass needs to add headers in the resulting splits.
+	 * If true, {@link #addHeaders(Message, AbstractIntegrationMessageBuilder)} will be called.
+	 * @param message the message.
+	 * @return true
+	 */
+	protected boolean willAddHeaders(Message<?> message) {
+		return false;
+	}
+
+	/**
+	 * Allows subclasses to add extra headers to the output messages. Headers may not be
+	 * removed by this method.
+	 *
+	 * @param message the inbound message.
+	 * @param headers the headers to add messages to.
+	 */
+	protected void addHeaders(Message<?> message, Map<String, Object> headers) {
```

Comment:
Although we can live with `Map` as well: we can't use Java 8 API, but end user can. Moreover its up to him what to do with `headers`.